pkg/oauth/tokenrequest/request_token.go

```diff
@@ -284,6 +291,71 @@ func (o *RequestTokenOptions) RequestToken() (string, error) {
 	}
 }
 
+func (o *RequestTokenOptions) RequestTokenWithAuthorization(authorizeUrlHandler func(string) error, callbackPort int) (string, error) {
+	if o.OsinConfig == nil {
+		if err := o.SetDefaultOsinConfig(); err != nil {
```

Comment:
you'll likely want this to accept the client id and the redirect URL

Reply:
See https://github.com/openshift/library-go/pull/1531/files#diff-f30259f7a13841a226555f2801678f096da8bb80456cf414201b8cfd647da728R137